### PR TITLE
handle locale when null

### PR DESCRIPTION
**Pull request by carl-textalk**

### CODE REVIEW

File: test/org/daisy/dotify/formatter/impl/segment/TextSegmentTest.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+public class TextSegmentTest {
+    @Test
+   public void testGetLocaleReturnEmptyOptionalIfLocaleIsNull() {
+        // Test that verifies that getLocale does not crash if locale is null.
+        TextProperties.Builder builder = new TextProperties.Builder(null);
+        TextProperties tp = builder.build();
+        String chars = "";
+        TextSegment ts = new TextSegment(chars, tp, false);
+        Optional <String> locale = ts.getLocale();
```
Comment (carl-textalk) [inline test/org/daisy/dotify/formatter/impl/segment/TextSegmentTest.java]: `ts.GetLocal` did throw a NullPointerException before .

File: test/org/daisy/dotify/formatter/impl/segment/TextSegmentTest.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        assertEquals(Optional.empty(), locale);
+    }
+
+    @Test
+    public void testVerifyComparisonWIthOptionals() {
```
Comment (carl-textalk) [inline test/org/daisy/dotify/formatter/impl/segment/TextSegmentTest.java:27]: This test is just to verify the behaviour when comparing the returned value from getLocal.

**Comment by bertfrees** [comment]
The question is: is locale supposed to be null? Maybe this should be caught earlier? When validating the OBFL, or when building the `TextProperties`? Note that is also the locale "und" (undetermined), and some other locales to express a deviation from the normal case ("zxx", "mis", "mul", ...).

**Comment by carl-textalk** [comment]
@PaulRambags @bertfrees Are you okay with these changes?

**Review by bertfrees — APPROVED** [review]

**Review by PaulRambags — APPROVED** [review]

**Comment by kalaspuffar** [comment]
Have run the regression tests without any issues and will merge this PR.

**Pull request merged**